src/shapes/Textbox.ts

```diff
@@ -306,6 +310,58 @@ export class Textbox extends IText {
     return value.split(this._wordJoiners);
   }
 
+  measureLineForWrapping(line: string, lineIndex: number) {
+    const splitByGrapheme = this.splitByGrapheme,
+      parts = splitByGrapheme ? this.graphemeSplit(line) : this.wordSplit(line);
+
+    if (parts.length === 0) {
+      return null;
+    }
+
+    const { data, minWidth } = parts.reduce(
+      ({ data, offset, minWidth }, part, index) => {
+        let infixWidth = 0;
+        let infix: string | null = null;
+        if (index > 0 && !splitByGrapheme) {
+          infix = ' ';
+          // measure infix at offset to respect styling etc.
+          infixWidth = this._measureWord(infix, lineIndex, offset);
+          // move cursor after infix
+          offset += infix.length;
+        }
+        // split words if necessary
+        const graphemes = !splitByGrapheme
+          ? this.graphemeSplit(part)
+          : // we must use concat for compatibility
+            [].concat(part);
```

Comment:
not sure I'm understanding this. Why not just use `[ part ]`?

Reply:
I tried that.
some tests were failing but like this they were not.
It is really strange

Reply:
if part is an array, [].concat(part) is part and not [part]
I think that is the the difference.